core/src/main/java/io/ddf/DDFCache.java

```diff
@@ -94,14 +170,19 @@ public DDF getDDFByName(String name) throws DDFException {
   }
 
   public synchronized void setDDFName(DDF ddf, String name) throws DDFException {
-    if(!Strings.isNullOrEmpty(name)) {
-      if(!Strings.isNullOrEmpty(ddf.getName())) {
-        this.mUris.remove(ddf.getUri());
+    synchronized (ddf) {
```

Comment:
This class has several granularities of lock. Can we have a short description of its locking strategy at the beginning?